### DISCUSSION TITLE
Excessive totalsize in trace_d_b task_execution_d_b collections in Mongo

### DISCUSSION THREAD

**Discussion by carlosverbo** [StackStorm/st2#5623] [Q&A]
Hi,

I have huge totalSize for collections trace_d_b and task_execution_d_b, as result,  fs occupation reached 93%, I need to decrease it.

After purge-executions and purge-trigger and compact, totalSize doesn't decrease. 

/opt/stackstorm/st2/bin/st2-purge-executions --config-file /etc/st2/st2.conf --timestamp="2022-04-14T00:00:00.000000Z" --purge-incomplete
/opt/stackstorm/st2/bin/st2-purge-trigger-instances --config-file /etc/st2/st2.conf --timestamp="2022-04-141T00:00:00.000000Z"

> db.trace_d_b.totalSize()
11010805760
> db.task_execution_d_b.totalSize()
8700403712

any procedure to realase space in mentioned collections?

Thanks in advance

**Comment by amanda11** [comment]
Hi,
There are new garbage collection options added in StackStorm 3.7.0 to address this issue, but you cannot  upgrade to 3.7 and then  either run the new scripts or configure the automatic garbage colleciton - see https://docs.stackstorm.com/troubleshooting/purging_old_data.html

If you can't upgrade, then you'd need to do a manual purge on the collections with the appropriate mongo commands (you can take a look at the underlying garbage collection code in github to see what  queries we use for purging).
